File: 151051/Maliuk/kr/src/main/java/ru/bsuir/mapper/CreatorMapper.java
package ru.bsuir.mapper;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import ru.bsuir.dto.request.CreatorRequestTo;
import ru.bsuir.dto.response.CreatorResponseTo;
import ru.bsuir.entity.Creator;

@Mapper(componentModel = "spring")
public interface CreatorMapper {

    @Mapping(target = "id", ignore = true)
    Creator toEntity(CreatorRequestTo creatorRequest);

    CreatorResponseTo toDTO(Creator creator);
}

File: 251002/Nikonorova/labs3-5/publisher/Mappers/CreatorMapper.cs
using Microsoft.AspNetCore.Http.HttpResults;
using publisher.DTO.Request;
using publisher.DTO.Response;
using Riok.Mapperly.Abstractions;
using publisher.Models;

namespace publisher.Mappers
{
    [Mapper]
    public static partial class CreatorMapper
    {
        public static partial Creator ToEntity(this CreateCreatorRequestTo createCreatorRequestTo);
        public static partial CreatorResponseTo ToResponse(this Creator creator);
        public static partial IEnumerable<CreatorResponseTo> ToResponse(this IEnumerable<Creator> creators);
    }
}
